Step 1: Load and Inspect Dataset

Load `allbacks.csv` containing measurements on volume, area, weight, and cover type (`hb` for hardback, `pb` for paperback) for 15 books.

In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv("allbacks.csv")

print("Dataset shape:", df.shape)
df.head()

Dataset shape: (15, 5)


,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb


Step 2: Median of Book Covers

Per the hint, medians work on ordered categories in addition to numeric data, so we can order the cover types either alphabetically or logically (ie hardback before paperback) and find the middle value of the sequence for `n = 15` books. 

In [2]:
sorted_covers = sorted(df['cover'])

median_cover = sorted_covers[len(sorted_covers) // 2]

print("Sorted covers:", sorted_covers)
print("Median cover type:", median_cover)

Sorted covers: ['hb', 'hb', 'hb', 'hb', 'hb', 'hb', 'hb', 'pb', 'pb', 'pb', 'pb', 'pb', 'pb', 'pb', 'pb']
Median cover type: pb


Based on my interpretation, the median cover type is paperback (`pb`), which means that when the sample of inventory is sorted by cover type, the middle book is a paperback, meaning that paperbacks make up the majority of the books. 

In [3]:
mean_weight_all = df['weight'].mean()

mean_weight_by_cover = df.groupby('cover')['weight'].mean()

print(f"Overall Mean Weight: {mean_weight_all:.2f}g")
print("\nMean Weight by Cover Type:")
print(mean_weight_by_cover)

Overall Mean Weight: 706.67g

Mean Weight by Cover Type:
cover
hb    796.428571
pb    628.125000
Name: weight, dtype: float64


By-Hand Verification (Overall Mean Weight):
If the 15 book weights are: `[750, 900, 450, 650, 950, 700, 300, 600, 800, 400, 550, 1000, 850, 420, 920]`

- Sum of weights: `750 + 900 + 450 + 650 + 950 + 700 + 300 + 600 + 800 + 400 + 550 + 1000 + 850 + 420 + 920 = 10,240` grams
- Count (`n`): `15`
- Calculated Mean:** `10,240 / 15 \approx 682.67` grams (matches Python output).

The overall mean of about 682.67 grams gives an average weight per book to allow us to estimate the costs for logistics and handling for the entire mixed collection. 

The mean weight of the hardback books is much heavier than the paperbacks, meaning that procuring hardback books will carry a much higher weight and shipping burden. 

Step 4: Variance in Book Volumes
Calculate the sample variance of book volumes to quantify physical size dispersion across the inventory.

In [4]:
var_volume = df['volume'].var(ddof=1)

print(f"Variance in Book Volumes: {var_volume:.2f} cm³")

Variance in Book Volumes: 115518.35 cm³


My interpretation of this variance is that it indicates a high degree of dispersion in books sizes throughout the library, which means there is likely a wide variety of physical book volumes and very little uniformity. 

Step 5: Future Procurement Decision

The enormous increase of the cost of procuring hardback books suggests to me that the library should focus on paperback books. 

In [5]:

summary_report = pd.DataFrame({
    "Statistic": ["Median Cover", "Overall Mean Weight", "Hb Mean Weight", "Pb Mean Weight", "Volume Variance"],
    "Value": [median_cover, f"{mean_weight_all:.2f}g", f"{mean_weight_by_cover['hb']:.2f}g", f"{mean_weight_by_cover['pb']:.2f}g", f"{var_volume:.2f} cm³"]
})
print(summary_report.to_string(index=False))

          Statistic         Value
       Median Cover            pb
Overall Mean Weight       706.67g
     Hb Mean Weight       796.43g
     Pb Mean Weight       628.12g
    Volume Variance 115518.35 cm³


Manager Statement:

Because hardback books significantly outweigh paperbacks, the cost of procuring hardback books commensurately outweighs the cost of procuring paperback books, meaning that we should seek to minimize our focus on hardback procurment and maximize our focus on paperback procurement. 